lib/src/main/java/io/ably/lib/realtime/AblyRealtime.java

```diff
@@ -72,10 +72,19 @@ public void close() {
 	public class Channels extends HashMap<String, Channel> {
 		public Channels() {
 			/* remove all channels when the connection is closed, to avoid stalled state */
-			connection.on(ConnectionState.closed, new ConnectionStateListener() {
+			connection.on(new ConnectionStateListener() {
```

Comment:
This was also a problem with what was there before so probably can be fixed in a new PR, but if you call `off()` (without arguments) on the `Connection` ([which calls this](https://github.com/ably/ably-java/blob/master/lib/src/main/java/io/ably/lib/util/EventEmitter.java#L24-L27)) this listener will be removed and thus this logic will be broken. I fixed this in iOS [by having an internal EventEmitter](https://github.com/ably/ably-ios/blob/master/Source/ARTRealtime.m#L68).